js/piwik.js

```diff
@@ -2442,17 +2442,59 @@ if (typeof Piwik !== 'object') {
 
                 return false;
             }
+            /**
+             * Build a compatibility callback system for piwik
+             * @param  {Function} cb Can be an object too for newer version of piwik
+             * @return {Object}      {success,error}
+             */
+            function getCallBacksRequest(cb) {
+                /*property
+                    success,error
+                */
+
+                // prevent jslint error "Unexpected /*property*/ 'success'."
+                var _callBack = {
+                    success: function () {},
+                    error: function () {}
+                };
+
+                if(cb) {
+                    // compatibility with older piwik
+                    if ('function' === typeof cb) {
+                        _callBack.success = _callBack.error = cb;
+                    }
+
+                    if ('object' === typeof cb) {
+
+                        if ('function' === typeof cb.success) {
+                            _callBack.success = cb.success;
+                        }
+
+                        if ('function' === typeof cb.error) {
+                            _callBack.error = cb.error;
+                        }
+
+                    }
+                }
+                return _callBack;
+            }
 
             /*
              * Send image request to Piwik server using GET.
              * The infamous web bug (or beacon) is a transparent, single pixel (1x1) image
              */
             function getImage(request, callback) {
+                var cb = getCallBacksRequest(callback);
                 var image = new Image(1, 1);
 
                 image.onload = function () {
                     iterator = 0; // To avoid JSLint warning of empty block
-                    if (typeof callback === 'function') { callback(); }
+                    cb.success();
+                };
+
+                image.onerror = function () {
```

Comment:
this callback will be called when we have other than HTTP 200, even it is HTTP 204 no content.